webui/src/lib/remark-plugins/imageUriReplacer.ts

```diff
@@ -0,0 +1,33 @@
+import { visit } from "unist-util-visit";
+import type { Node } from "unist";
+import type { Root } from "mdast";
+import type { Plugin } from "unified";
+
+import { objects } from "../api";
+
+const imageUriReplacer: Plugin<[], Root> = () => async (tree) => {
+  const images: Array<Node & { url: string }> = [];
+  visit(tree, "image", (node: Node & { url: string }) => {
```

Comment:
Isn't `img` a valid image label too?

Reply:
In markdown, an image node is `![alt](URL)`. Not sure if this AST matches HTML `<img />` tags as well. I'll need to check.